frontend/src/components/CreateLocation/CreateLocation.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Input,
+  TextareaAutosize,
+  Typography,
+  Button,
+  Autocomplete,
+  TextField
+} from '@mui/material';
+import axios from 'axios';
+import React, { useState, useRef } from 'react';
+import { latlngType } from '../../../types';
+import { getFiltersForUser } from '../../static/mainFIlters';
+
+type Props = {
+  coordinate: latlngType;
+};
+
+const CreateLocation = ({ coordinate }: Props) => {
+  const ref = useRef<null | HTMLInputElement>();
+  const [filters, setFilters] = useState('');
+
+  const [locationName, setLocationName] = useState('');
+  const [description, setDescription] = useState('');
+  const [, setLinks] = useState<string[]>([]);
+  const [files, setFiles] = useState<File[]>([]);
+
+  const proces = process.env.REACT_APP_API_URI;
+
+  const handleChange = (e: any): void => {
+    setLocationName(e.target.value);
+  };
+
+  const handleChangeDescription = (e: any): void => {
+    setDescription(e.target.value);
+  };
+
+  const accessToken = localStorage.getItem('accessToken');
+
+  const onSubmit = async (e: any) => {
+    e.preventDefault();
+    try {
+      const formData = new FormData();
+      formData.append('locationName', locationName);
+      formData.append('description', description);
+      formData.append('coordinates', String(coordinate.lat));
+      formData.append('coordinates', String(coordinate.lng));
+      formData.append('filters', String(filters));
+      formData.append('image', files[0]);
+
+      await axios.post(`${proces}add_personal_location`, formData, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+          'Content-Type': 'multipart/form-data'
+        }
+      });
+
+      setLocationName('');
+      setDescription('');
+      if (ref.current) {
+        ref.current.value = '';
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const onChangeAutocomlete = (e: any, value: any) => {
+    setFilters(value);
+  };
+
+  const handleFilesChange = (e: any) => {
+    const { files: filesLst } = e.currentTarget;
+    const filesListArr = [];
+
+    if (filesLst) {
+      for (let i = 0; i < filesLst.length; i += 1) {
+        filesListArr.push(filesLst[i]);
+      }

```

Comment:
just why? :)

Reply:
this code i copy and paste from Yuriy  - this is need for images